mtest/src/PipeTest.cxx

```diff
@@ -1580,13 +1608,31 @@ namespace mtest {
   }  // end of computeIntegralValue
 
   real PipeTest::computeMeanValue(const StudyCurrentState& state,
-                                  const std::string& n) const {
+                                  const std::string& n,
+                                  const Configuration c) const {
     constexpr real pi = 3.14159265358979323846;
-    const auto ri = this->mesh.inner_radius;
-    const auto re = this->mesh.outer_radius;
-    const auto h = 1;
+    const auto ri = [this, state, c]() -> real {
+      if ((!this->hpp) && (c == Configuration::CURRENT_CONFIGURATION)) {
+        return state.u1[0];
+      }
+      return this->mesh.inner_radius;
+    }();
+    const auto re = [this, state, c]() -> real {
+      if ((!this->hpp) && (c == Configuration::CURRENT_CONFIGURATION)) {
+        const size_type ln = this->getNumberOfNodes() - 1;
+        const auto Re = this->mesh.outer_radius;
+        return (this->hpp) ? Re : Re + state.u1[ln];
```

Comment:
shall be return Re+....
(testing this->hpp is meaningless here)

Reply:
see the commit https://github.com/thelfer/tfel/pull/252/commits/407a9bc65dca6aca5d901b219889fc36e5599ba4